File: client/src/data/litters.ts
export interface Litter {
  id: string;
  name: string;
  mother_id?: string;
  mother_name?: string;
  father_id?: string;
  father_name?: string;
  birth_date: string;
  available_date: string;
  profile_picture_id: string;
  image_ids?: string;
  status: "Planned" | "Available" | "Sold";
}

export const mockLitters: Litter[] = [
  {
    id: "litter-1",
    name: "Litter 1",
    mother_id: "pug-2",
    father_id: "pug-1",
    birth_date: "2023-10-01",
    available_date: "2023-12-01",
    profile_picture_id: "img-litter-1",
    status: "Sold",
  },
  {
    id: "litter-2",
    name: "Litter 2",
    mother_id: "pug-3",
    father_name: "External Stud",
    birth_date: "2024-02-15",
    available_date: "2024-04-15",
    profile_picture_id: "img-litter-2",
    image_ids: "img-litter-2,img-pup-1,img-pup-2",
    status: "Available",
  },
  {
    id: "litter-3",
    name: "Litter 3",
    mother_id: "pug-2",
    father_id: "pug-1",
    birth_date: "2024-05-20",
    available_date: "2024-07-20",
    profile_picture_id: "img-litter-3",
    status: "Planned",
  },
];

File: server/internal/models/litters.go
package models

import "time"

type Litter struct {
	ID             int       `json:"id" db:"id"`
	Name           string    `json:"name" form:"name" binding:"required" db:"name"`
	MotherID       *int      `json:"mother_id" db:"mother_id"`
	FatherID       *int      `json:"father_id" db:"father_id"`
	ExternalMother string    `json:"external_mother_name" form:"external_mother_name" db:"external_mother_name"`
	ExternalFather string    `json:"external_father_name" form:"external_father_name" db:"external_father_name"`
	MotherName     string    `json:"mother_name" db:"-"` 
	FatherName     string    `json:"father_name" db:"-"` 
	BirthDate      time.Time `json:"birth_date" form:"birth_date" db:"birth_date"`
	AvailableDate  time.Time `json:"available_date" form:"available_date" db:"available_date"`
	Status         string    `json:"status" form:"status" db:"status"`
	ProfilePicture *Image 	 `json:"profilePicture,omitempty" db:"profile_picture"`
	Gallery 			 []Image 	 `json:"gallery,omitempty" db:"gallery"`
	CreatedAt      time.Time `json:"createdAt" db:"created_at"`
	UpdatedAt      time.Time `json:"updatedAt" db:"updated_at"`
}
